BIGDreamBooth/executor/executor.py

```diff
@@ -459,31 +459,21 @@ def generate(self, docs: DocumentArray, parameters: Dict = None, **kwargs):
 
     def _generate(self, num_images: int, model_path: str, prompt: str, batch_size: int, revision=None) -> DocumentArray:
         # call scripts/generate.py
-        cur_dir = os.path.abspath(os.path.join(__file__, '..'))
+        cur_dir = os.path.dirname(os.path.abspath(__file__))
         with tempfile.TemporaryDirectory() as tmp_dir:
-            if self.is_colab:
-                cmd_args = [
-                    'python',
-                    os.path.join(cur_dir, 'scripts/generate.py'),
-                    '--save_dir', tmp_dir,
-                    '--model_path', model_path,
-                    '--prompt', prompt,
-                    '--num_images', str(num_images),
-                    '--batch_size', str(batch_size),
-                ]
-                if revision:
-                    cmd_args += ['--revision', revision]
-                output, err = cmd(cmd_args)
-                handle_error_messages_from_cmd([output, err], cmd_args)
-            else:
-                from .scripts.generate import _generate
-                _generate(
-                    save_dir=tmp_dir,
-                    model_path=model_path,
-                    prompt=prompt,
-                    num_images=num_images,
-                    batch_size=batch_size,
-                )
+            cmd_args = [
+                'python',
```

Comment:
just a comment, why u use a command and not expose the library equivalent to the main entrypoint? you are creating a new process which is expensive and can yield to more problems

Reply:
What do you mean with `expose the library equivalent to the main entrypoint`?
The issue I am tackling with calling it in a subprocess is that `pipeline = StableDiffusionPipeline.from_pretrained(...)` leads to high RAM usage even after we call
```
del pipeline
gc.collect()
if torch.cuda.is_available():
    torch.cuda.empty_cache()
```
The issue is a memory leak which would make it unusable for Colab to constantly allocate this high RAM.